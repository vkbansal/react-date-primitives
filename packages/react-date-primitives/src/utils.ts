export const DayOfMonthSymbol = Symbol('DayOfMonth');
export const DayOfRangeMonthSymbol = Symbol('DayOfRangeMonth');

export enum DayName {
    MONDAY = 'MONDAY',
    TUESDAY = 'TUESDAY',
    WEDNESDAY = 'WEDNESDAY',
    THURSDAY = 'THURSDAY',
    FRIDAY = 'FRIDAY',
    SATURDAY = 'SATURDAY',
    SUNDAY = 'SUNDAY'
}

export const DaysOfTheWeek: DayName[] = [
    DayName.SUNDAY,
    DayName.MONDAY,
    DayName.TUESDAY,
    DayName.WEDNESDAY,
    DayName.THURSDAY,
    DayName.FRIDAY,
    DayName.SATURDAY
];

type Mutable<T> = {
    -readonly [P in keyof T]: T[P];
};

export interface DayOfMonth {
    readonly __type: typeof DayOfMonthSymbol;
    readonly dateObj: Date;
    readonly dayName: DayName;
    readonly inCurrentMonth: boolean;
    readonly ISODateString: string;
}

export interface DayOfRangeMonth extends Omit<DayOfMonth, '__type'> {
    readonly __type: typeof DayOfRangeMonthSymbol;
    readonly inRange: boolean;
    readonly isStart: boolean;
    readonly isEnd: boolean;
}

export interface Month {
    readonly days: DayOfMonth[];
    readonly daysOfWeek: DayName[];
    readonly month: Date;
}

export interface RangeMonth {
    readonly days: DayOfRangeMonth[];
    readonly month: Date;
}

export interface RangeMonths {
    readonly months: RangeMonth[];
    readonly daysOfWeek: DayName[];
}

const monthsCache = new Map<string, Month>();
const rangeMonthsCache = new Map<string, RangeMonths>();

export function addDays(date: Date, days: number): Date {
    return setDay(date, date.getDate() + days);
}

export function addMonths(date: Date, months: number): Date {
    return setMonth(date, date.getMonth() + months);
}

export function setDay(date: Date, day: number): Date {
    const newDate = new Date(date.getTime());

    newDate.setDate(day);

    return newDate;
}

export function setMonth(date: Date, month: number): Date {
    const newDate = new Date(date.getTime());

    newDate.setMonth(month);

    return newDate;
}

export function setYear(date: Date, year: number): Date {
    const newDate = new Date(date.getTime());

    newDate.setFullYear(year);

    return newDate;
}

export function startOfMonth(date: Date): Date {
    const newDate = new Date(date.getTime());

    newDate.setDate(1);
    newDate.setHours(0, 0, 0, 0);

    return newDate;
}

export function endOfMonth(date: Date): Date {
    const newDate = new Date(date.getTime());

    newDate.setFullYear(date.getFullYear(), date.getMonth() + 1, 0);
    newDate.setHours(23, 59, 59, 999);

    return newDate;
}

export function isSameMonth(dateLeft: Date, dateRight: Date): boolean {
    return (
        dateLeft.getFullYear() === dateRight.getFullYear() &&
        dateLeft.getMonth() === dateRight.getMonth()
    );
}

export function isSameDay(dateLeft: Date, dateRight: Date): boolean {
    return (
        dateLeft.getFullYear() === dateRight.getFullYear() &&
        dateLeft.getMonth() === dateRight.getMonth() &&
        dateLeft.getDate() === dateRight.getDate()
    );
}

export function isDayAfter(date: Date, dateToCompare: Date): boolean {
    const newDate = new Date(date.getTime());
    const newDateToCompare = new Date(dateToCompare.getTime());

    newDate.setHours(0, 0, 0, 0);
    newDateToCompare.setHours(0, 0, 0, 0);

    return newDate.getTime() > newDateToCompare.getTime();
}

export function isDayBefore(date: Date, dateToCompare: Date): boolean {
    const newDate = new Date(date.getTime());
    const newDateToCompare = new Date(dateToCompare.getTime());

    newDate.setHours(0, 0, 0, 0);
    newDateToCompare.setHours(0, 0, 0, 0);

    return newDate.getTime() < newDateToCompare.getTime();
}

export function toISODateString(date: Date): string {
    const year = date.getFullYear();
    const month = (date.getMonth() + 1).toString().padStart(2, '0');
    const day = (date.getDate() + 0).toString().padStart(2, '0');

    return `${year}-${month}-${day}`;
}

export function callIfExists<T extends unknown[], U = unknown>(
    func: ((...args: T) => U) | undefined | null,
    ...args: T
): U | undefined {
    if (typeof func === 'function') {
        return func(...args);
    }
}

export function getDaysOfMonth(month: Date, weekStartsOn = DayName.SUNDAY): Month {
    const firstDate = startOfMonth(month);
    const dayOffset = DaysOfTheWeek.findIndex((d) => d === weekStartsOn);
    const key = `${toISODateString(firstDate)}-${weekStartsOn}`;

    if (dayOffset < 0) {
        throw new RangeError(`Invalid value for day provided: ${weekStartsOn}`);
    }

    const data = monthsCache.get(key);

    if (data) {
        return data;
    }

    const firstDayInCurrentMonth = firstDate.getDay();

    const days: Month['days'] = Array.from(
        { length: 6 * 7 },
        (_1, i): DayOfMonth => {
            const day = addDays(
                firstDate,
                i -
                    firstDayInCurrentMonth -
                    (dayOffset > firstDayInCurrentMonth ? 7 : 0) +
                    dayOffset
            );

            return {
                __type: DayOfMonthSymbol,
                dateObj: day,
                dayName: DaysOfTheWeek[(i + dayOffset) % 7],
                inCurrentMonth: isSameMonth(firstDate, day),
                ISODateString: toISODateString(day)
            };
        }
    );

    const daysOfWeek = DaysOfTheWeek.slice(dayOffset).concat(DaysOfTheWeek.slice(0, dayOffset));

    monthsCache.set(key, { days, daysOfWeek, month: firstDate });

    return { days, daysOfWeek, month: firstDate };
}

export function getDaysOfRangeMonth(
    rangeMonths: Date[],
    startDate?: Date | null,
    endDate?: Date | null,
    weekStartsOn = DayName.SUNDAY
): RangeMonths {
    let key = rangeMonths
        .map((month): string => {
            if (!(month instanceof Date)) {
                throw new TypeError('Given month is not an instance of Date');
            }
            return `${month.getFullYear()}-${month.getMonth()}`;
        })
        .join('|');

    key += `|${startDate ? toISODateString(startDate) : ''}`;
    key += `|${endDate ? toISODateString(endDate) : ''}`;
    key += `|${weekStartsOn}`;

    if (rangeMonthsCache.has(key)) {
        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
        return rangeMonthsCache.get(key)!;
    }

    const months = rangeMonths.reduce<Mutable<RangeMonths>>(
        (accumulator, currentMonth, i) => {
            const { days, daysOfWeek } = getDaysOfMonth(currentMonth, weekStartsOn);

            if (i === 0) {
                accumulator.daysOfWeek = daysOfWeek.slice(0);
            }

            const daysOfCurrentMonth = days.map(
                (day): DayOfRangeMonth => {
                    const isStart = Boolean(startDate && isSameDay(day.dateObj, startDate));
                    const isEnd = Boolean(endDate && isSameDay(day.dateObj, endDate));

                    return {
                        ...day,
                        __type: DayOfRangeMonthSymbol,
                        inRange:
                            isStart ||
                            isEnd ||
                            Boolean(
                                startDate &&
                                    isDayAfter(day.dateObj, startDate) &&
                                    endDate &&
                                    isDayBefore(day.dateObj, endDate)
                            ),
                        isStart,
                        isEnd
                    };
                }
            );

            accumulator.months.push({
                days: daysOfCurrentMonth,
                month: startOfMonth(currentMonth)
            });

            return accumulator;
        },
        { months: [], daysOfWeek: [] }
    );

    rangeMonthsCache.set(key, months);

    return months as RangeMonths;
}
