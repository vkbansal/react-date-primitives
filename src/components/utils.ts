import { DayOfMonthSubset } from './types';

const monthsCache: Map<string, DayOfMonthSubset[][]> = new Map();

export function setDay(date: Date, day: number): Date {
    const newDate = new Date(date.getTime());

    newDate.setDate(day);

    return newDate;
}

export function addDays(date: Date, days: number): Date {
    return setDay(date, date.getDate() + days);
}

export function setMonth(date: Date, month: number): Date {
    const newDate = new Date(date.getTime());

    newDate.setMonth(month);

    return newDate;
}

export function addMonths(date: Date, months: number): Date {
    return setMonth(date, date.getMonth() + months);
}

export function setYear(date: Date, year: number): Date {
    const newDate = new Date(date.getTime());

    newDate.setFullYear(year);

    return newDate;
}

export function startOfMonth(date: Date): Date {
    const newDate = new Date(date.getTime());

    newDate.setDate(1);
    newDate.setHours(0, 0, 0, 0);

    return newDate;
}

export function endOfMonth(date: Date): Date {
    const newDate = new Date(date.getTime());

    newDate.setFullYear(date.getFullYear(), date.getMonth() + 1, 0);
    newDate.setHours(23, 59, 59, 999);

    return newDate;
}

export function isSameMonth(dateLeft: Date, dateRight: Date): boolean {
    return (
        dateLeft.getFullYear() === dateRight.getFullYear() &&
        dateLeft.getMonth() === dateRight.getMonth()
    );
}

export function isSameDay(dateLeft: Date, dateRight: Date): boolean {
    return (
        dateLeft.getFullYear() === dateRight.getFullYear() &&
        dateLeft.getMonth() === dateRight.getMonth() &&
        dateLeft.getDate() === dateRight.getDate()
    );
}

export function isDayAfter(date: Date, dateToCompare: Date): boolean {
    const newDate = new Date(date.getTime());
    const newDateToCompare = new Date(dateToCompare.getTime());

    newDate.setHours(0, 0, 0, 0);
    newDateToCompare.setHours(0, 0, 0, 0);

    return newDate.getTime() > newDateToCompare.getTime();
}

export function isDayBefore(date: Date, dateToCompare: Date): boolean {
    const newDate = new Date(date.getTime());
    const newDateToCompare = new Date(dateToCompare.getTime());

    newDate.setHours(0, 0, 0, 0);
    newDateToCompare.setHours(0, 0, 0, 0);

    return newDate.getTime() < newDateToCompare.getTime();
}

export function callIfExists(callback?: any, ...args: any[]) {
    if (typeof callback === 'function') {
        callback(...args);
    }
}

export function getDatesofMonth(month: Date): DayOfMonthSubset[][] {
    const firstDate = startOfMonth(month);
    const key = `${firstDate.getFullYear()}-${firstDate.getMonth()}`;

    const data = monthsCache.get(key);

    if (data) {
        return data;
    }

    const firstDayInCurrentMonth = firstDate.getDay();

    const newdata: DayOfMonthSubset[][] = Array.from(
        { length: 6 },
        (_1, i): DayOfMonthSubset[] => {
            const week = Array.from(
                { length: 7 },
                (_2, j): DayOfMonthSubset => {
                    const day = addDays(firstDate, i * 7 + j - firstDayInCurrentMonth);

                    return { date: day, inCurrentMonth: isSameMonth(firstDate, day) };
                }
            );

            return week;
        }
    );

    monthsCache.set(key, newdata);

    return newdata;
}
