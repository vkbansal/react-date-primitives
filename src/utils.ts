export interface DayOfMonth {
    date: Date;
    inCurrentMonth: boolean;
}

export type DaysOfMonth = DayOfMonth[][];

const monthsCache: Map<string, DaysOfMonth> = new Map();

export function setDay(date: Date, day: number): Date {
    const newDate = new Date(date.getTime());

    newDate.setDate(day);

    return newDate;
}

export function addDays(date: Date, days: number): Date {
    return setDay(date, date.getDate() + days);
}

export function setMonth(date: Date, month: number): Date {
    const newDate = new Date(date.getTime());

    newDate.setMonth(month);

    return newDate;
}

export function addMonths(date: Date, months: number): Date {
    return setMonth(date, date.getMonth() + months);
}

export function setYear(date: Date, year: number): Date {
    const newDate = new Date(date.getTime());

    newDate.setFullYear(year);

    return newDate;
}

export function startOfMonth(date: Date): Date {
    const newDate = new Date(date.getTime());

    newDate.setDate(1);
    newDate.setHours(0, 0, 0, 0);

    return newDate;
}

export function endOfMonth(date: Date): Date {
    const newDate = new Date(date.getTime());

    newDate.setFullYear(date.getFullYear(), date.getMonth() + 1, 0);
    newDate.setHours(23, 59, 59, 999);

    return newDate;
}

export function isSameMonth(dateLeft: Date, dateRight: Date): boolean {
    return (
        dateLeft.getFullYear() === dateRight.getFullYear() &&
        dateLeft.getMonth() === dateRight.getMonth()
    );
}

export function isSameDay(dateLeft: Date, dateRight: Date): boolean {
    return (
        dateLeft.getFullYear() === dateRight.getFullYear() &&
        dateLeft.getMonth() === dateRight.getMonth() &&
        dateLeft.getDate() === dateRight.getDate()
    );
}

export function isDayAfter(date: Date, dateToCompare: Date): boolean {
    const newDate = new Date(date.getTime());
    const newDateToCompare = new Date(dateToCompare.getTime());

    newDate.setHours(0, 0, 0, 0);
    newDateToCompare.setHours(0, 0, 0, 0);

    return newDate.getTime() > newDateToCompare.getTime();
}

export function isDayBefore(date: Date, dateToCompare: Date): boolean {
    const newDate = new Date(date.getTime());
    const newDateToCompare = new Date(dateToCompare.getTime());

    newDate.setHours(0, 0, 0, 0);
    newDateToCompare.setHours(0, 0, 0, 0);

    return newDate.getTime() < newDateToCompare.getTime();
}

export function callIfExists(callback?: any, ...args: any[]) {
    if (typeof callback === 'function') {
        callback(...args);
    }
}

export function getDaysOfMonth(month: Date): DayOfMonth[][] {
    const firstDate = startOfMonth(month);
    const key = `${firstDate.getFullYear()}-${firstDate.getMonth()}`;

    const data = monthsCache.get(key);

    if (data) {
        return data;
    }

    const firstDayInCurrentMonth = firstDate.getDay();

    const newdata: DaysOfMonth = Array.from(
        { length: 6 },
        (_1, i): DayOfMonth[] => {
            const week = Array.from(
                { length: 7 },
                (_2, j): DayOfMonth => {
                    const day = addDays(firstDate, i * 7 + j - firstDayInCurrentMonth);

                    return { date: day, inCurrentMonth: isSameMonth(firstDate, day) };
                }
            );

            return week;
        }
    );

    monthsCache.set(key, newdata);

    return newdata;
}
