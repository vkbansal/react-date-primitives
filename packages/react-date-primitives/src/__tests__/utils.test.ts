import * as utils from '../utils';

import MonthSerializer from './MonthSerializer';
import DateSerializer from './DateSerializer';

expect.addSnapshotSerializer(DateSerializer);
expect.addSnapshotSerializer(MonthSerializer);

const YEAR = 2021;

describe('utils test', () => {
  const month = new Date(YEAR, 0 /* Jan */, 1, 0, 0, 0, 0);

  afterEach(() => {
    expect(month).toBe(month); // immutability check
  });

  test('setDay', () => {
    const newDate = utils.setDay(month, 10);
    expect(newDate.getDate()).toBe(10);
    expect(newDate.getMonth()).toBe(month.getMonth());
    expect(newDate.getFullYear()).toBe(month.getFullYear());
  });

  test('addDays', () => {
    let newDate = utils.setDay(month, 10);
    newDate = utils.addDays(newDate, 10);
    expect(newDate.getDate()).toBe(20);
    expect(newDate.getMonth()).toBe(month.getMonth());
    expect(newDate.getFullYear()).toBe(month.getFullYear());
  });

  test('setMonth', () => {
    const newDate = utils.setMonth(month, 10);
    expect(newDate.getMonth()).toBe(10);
    expect(newDate.getDate()).toBe(month.getDate());
    expect(newDate.getFullYear()).toBe(month.getFullYear());
  });

  test('addMonths', () => {
    let newDate = utils.setMonth(month, 2);
    newDate = utils.addMonths(newDate, 3);
    expect(newDate.getMonth()).toBe(5);
    expect(newDate.getDate()).toBe(month.getDate());
    expect(newDate.getFullYear()).toBe(month.getFullYear());

    newDate = utils.addMonths(month, 14);
    expect(newDate.getMonth()).toBe(3);
    expect(newDate.getFullYear()).toBe(YEAR + 1);
  });

  test('setYear', () => {
    const newDate = utils.setYear(month, 2206);
    expect(newDate.getFullYear()).toBe(2206);
    expect(newDate.getDate()).toBe(month.getDate());
    expect(newDate.getMonth()).toBe(month.getMonth());
  });

  test('startOfMonth', () => {
    const date = new Date();
    const newDate = utils.startOfMonth(date);
    expect(newDate.getDate()).toBe(1);
    expect(newDate.getMonth()).toBe(date.getMonth());
    expect(newDate.getFullYear()).toBe(date.getFullYear());
  });

  test('endOfMonth', () => {
    const newDate = utils.endOfMonth(month);
    expect(newDate.getDate()).toBe(31);
    expect(newDate.getMonth()).toBe(month.getMonth());
    expect(newDate.getFullYear()).toBe(month.getFullYear());
  });

  test('isSameMonth', () => {
    expect(utils.isSameMonth(new Date(2014, 8 /* Sep */, 2), new Date(2014, 8 /* Sep */, 25))).toBe(
      true
    );

    expect(utils.isSameMonth(new Date(2014, 8 /* Sep */, 2), new Date(2013, 8 /* Sep */, 25))).toBe(
      false
    );
  });

  test('isSameDay', () => {
    expect(
      utils.isSameDay(new Date(2014, 8 /* Sep */, 4, 6, 0), new Date(2014, 8 /* Sep */, 4, 18, 0))
    ).toBe(true);

    expect(
      utils.isSameDay(new Date(2014, 8 /* Sep */, 4, 23, 59), new Date(2014, 8 /* Sep */, 5, 0, 0))
    ).toBe(false);
  });

  test('isDayAfter', () => {
    expect(utils.isDayAfter(new Date(1989, 6 /* Jul */, 10), new Date(1987, 1 /* Feb */, 11))).toBe(
      true
    );

    expect(utils.isDayAfter(new Date(1987, 1 /* Feb */, 11), new Date(1989, 6 /* Jul */, 10))).toBe(
      false
    );

    expect(utils.isDayAfter(new Date(1989, 6 /* Jul */, 10), new Date(1989, 6 /* Jul */, 10))).toBe(
      false
    );
  });

  test('isDayBefore', () => {
    expect(
      utils.isDayBefore(new Date(1987, 1 /* Feb */, 11), new Date(1989, 6 /* Jul */, 10))
    ).toBe(true);

    expect(
      utils.isDayBefore(new Date(1989, 6 /* Jul */, 10), new Date(1987, 1 /* Feb */, 11))
    ).toBe(false);

    expect(
      utils.isDayBefore(new Date(1989, 6 /* Jul */, 10), new Date(1989, 6 /* Jul */, 10))
    ).toBe(false);
  });

  test('callIfExists', () => {
    const callback = jest.fn();
    const args = [1, 2, 5, 'h'];

    expect(() => utils.callIfExists(undefined)).not.toThrow();
    utils.callIfExists(callback, ...args);
    expect(callback).toHaveBeenCalledWith(...args);
  });

  test('toISODateString', () => {
    expect(utils.toISODateString(new Date(1989, 6 /* Jul */, 10))).toBe('1989-07-10');
    expect(utils.toISODateString(new Date(1989, 1 /* Jul */, 11))).toBe('1989-02-11');
  });

  describe('getDaysOfMonth', () => {
    test('weeks start with SUNDAY by default', () => {
      const result = utils.getDaysOfMonth(month);
      expect(result.daysOfWeek[0]).toBe(utils.DayName.SUNDAY);
    });

    test('month obj has 42 days', () => {
      const result = utils.getDaysOfMonth(month);
      expect(result.days.length).toBe(42);
    });

    test.each(utils.DaysOfTheWeek)(`weeks start with %s`, (dayOfWeek) => {
      const result = utils.getDaysOfMonth(month, dayOfWeek);
      expect(result.daysOfWeek[0]).toBe(dayOfWeek);
      expect(result.days.length).toBe(42);
      expect(result).toMatchSnapshot();
    });
  });

  describe('getDaysOfRangeMonth', () => {
    test('works without startDate and endDate', () => {
      const months = Array.from({ length: 2 }, (_, i) => utils.addMonths(month, i));
      const result = utils.getDaysOfRangeMonth(months);
      expect(result.months[0]).toMatchSnapshot();
      expect(result.months[1]).toMatchSnapshot();
    });
  });
});
