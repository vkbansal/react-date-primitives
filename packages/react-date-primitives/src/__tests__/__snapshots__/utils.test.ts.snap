// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils test getDaysOfMonth weeks start with FRIDAY 1`] = `
|                Jan 2021                 |
|-----------------------------------------|
| FRI | SAT | SUN | MON | TUE | WED | THU |
|  01 |  02 |  03 |  04 |  05 |  06 |  07 |
|  08 |  09 |  10 |  11 |  12 |  13 |  14 |
|  15 |  16 |  17 |  18 |  19 |  20 |  21 |
|  22 |  23 |  24 |  25 |  26 |  27 |  28 |
|  29 |  30 |  31 |     |     |     |     |
|     |     |     |     |     |     |     |
`;

exports[`utils test getDaysOfMonth weeks start with MONDAY 1`] = `
|                Jan 2021                 |
|-----------------------------------------|
| MON | TUE | WED | THU | FRI | SAT | SUN |
|     |     |     |     |  01 |  02 |  03 |
|  04 |  05 |  06 |  07 |  08 |  09 |  10 |
|  11 |  12 |  13 |  14 |  15 |  16 |  17 |
|  18 |  19 |  20 |  21 |  22 |  23 |  24 |
|  25 |  26 |  27 |  28 |  29 |  30 |  31 |
|     |     |     |     |     |     |     |
`;

exports[`utils test getDaysOfMonth weeks start with SATURDAY 1`] = `
|                Jan 2021                 |
|-----------------------------------------|
| SAT | SUN | MON | TUE | WED | THU | FRI |
|     |     |     |     |     |     |  01 |
|  02 |  03 |  04 |  05 |  06 |  07 |  08 |
|  09 |  10 |  11 |  12 |  13 |  14 |  15 |
|  16 |  17 |  18 |  19 |  20 |  21 |  22 |
|  23 |  24 |  25 |  26 |  27 |  28 |  29 |
|  30 |  31 |     |     |     |     |     |
`;

exports[`utils test getDaysOfMonth weeks start with SUNDAY 1`] = `
|                Jan 2021                 |
|-----------------------------------------|
| SUN | MON | TUE | WED | THU | FRI | SAT |
|     |     |     |     |     |  01 |  02 |
|  03 |  04 |  05 |  06 |  07 |  08 |  09 |
|  10 |  11 |  12 |  13 |  14 |  15 |  16 |
|  17 |  18 |  19 |  20 |  21 |  22 |  23 |
|  24 |  25 |  26 |  27 |  28 |  29 |  30 |
|  31 |     |     |     |     |     |     |
`;

exports[`utils test getDaysOfMonth weeks start with THURSDAY 1`] = `
|                Jan 2021                 |
|-----------------------------------------|
| THU | FRI | SAT | SUN | MON | TUE | WED |
|     |  01 |  02 |  03 |  04 |  05 |  06 |
|  07 |  08 |  09 |  10 |  11 |  12 |  13 |
|  14 |  15 |  16 |  17 |  18 |  19 |  20 |
|  21 |  22 |  23 |  24 |  25 |  26 |  27 |
|  28 |  29 |  30 |  31 |     |     |     |
|     |     |     |     |     |     |     |
`;

exports[`utils test getDaysOfMonth weeks start with TUESDAY 1`] = `
|                Jan 2021                 |
|-----------------------------------------|
| TUE | WED | THU | FRI | SAT | SUN | MON |
|     |     |     |  01 |  02 |  03 |  04 |
|  05 |  06 |  07 |  08 |  09 |  10 |  11 |
|  12 |  13 |  14 |  15 |  16 |  17 |  18 |
|  19 |  20 |  21 |  22 |  23 |  24 |  25 |
|  26 |  27 |  28 |  29 |  30 |  31 |     |
|     |     |     |     |     |     |     |
`;

exports[`utils test getDaysOfMonth weeks start with WEDNESDAY 1`] = `
|                Jan 2021                 |
|-----------------------------------------|
| WED | THU | FRI | SAT | SUN | MON | TUE |
|     |     |  01 |  02 |  03 |  04 |  05 |
|  06 |  07 |  08 |  09 |  10 |  11 |  12 |
|  13 |  14 |  15 |  16 |  17 |  18 |  19 |
|  20 |  21 |  22 |  23 |  24 |  25 |  26 |
|  27 |  28 |  29 |  30 |  31 |     |     |
|     |     |     |     |     |     |     |
`;

exports[`utils test getDaysOfRangeMonth works without startDate and endDate 1`] = `
|                Jan 2021                 |
|-----------------------------------------|
|
|     |     |     |     |     |  01 |  02 |
|  03 |  04 |  05 |  06 |  07 |  08 |  09 |
|  10 |  11 |  12 |  13 |  14 |  15 |  16 |
|  17 |  18 |  19 |  20 |  21 |  22 |  23 |
|  24 |  25 |  26 |  27 |  28 |  29 |  30 |
|  31 |     |     |     |     |     |     |
`;

exports[`utils test getDaysOfRangeMonth works without startDate and endDate 2`] = `
|                Feb 2021                 |
|-----------------------------------------|
|
|     |  01 |  02 |  03 |  04 |  05 |  06 |
|  07 |  08 |  09 |  10 |  11 |  12 |  13 |
|  14 |  15 |  16 |  17 |  18 |  19 |  20 |
|  21 |  22 |  23 |  24 |  25 |  26 |  27 |
|  28 |     |     |     |     |     |     |
|     |     |     |     |     |     |     |
`;
